internal/mithril-metric/src/helper.rs

```diff
@@ -0,0 +1,233 @@
+/// Create a MetricService.
+///
+/// To build the service you need to provide the structure name and a list of metrics.
+/// Each metrics is defined by an attribute name, a type, a metric name and a help message.
+///
+/// The attribute name will be used to create a getter method for the metric.
+///
+/// Crate that use this macro should have `paste` as dependency.
+///
+/// build_metrics_service!(
+///     MetricsService,
+///     counter_example: MetricCounter(
+///         "custom_counter_example_name",
+///         "Example of a counter metric"
+///     ),
+///     gauge_example: MetricGauge(
+///         "custom_gauge_example_name",
+///         "Example of a gauge metric"
+///     )
+/// );
+///
+/// let service = MetricsService::new(TestLogger::stdout()).unwrap();
+/// service.get_counter_example().record();
+/// service.get_gauge_example().record(Epoch(12));
+#[macro_export]
+macro_rules! build_metrics_service {
+    ($service:ident, $($metric_attribute:ident:$metric_type:ident($name:literal, $help:literal)),*) => {
+        paste::item! {
+            /// Metrics service which is responsible for recording and exposing metrics.
+            pub struct $service {
+                registry: Registry,
```

Comment:
You should use fully qualified path inside macro so their user don't need to import the dependencies used by the macro.
```suggestion
                registry: prometheus::Registry,
```